src/java.base/share/classes/java/lang/reflect/MemberPattern.java

```diff
@@ -407,14 +408,14 @@ public Object[] invoke(Object matchCandidate)
             Method method = getDeclaringClass().getDeclaredMethod(underlyingName, matchCandidate.getClass(), MethodHandle.class);
             method.setAccessible(override);
             MethodType bindingMT = MethodType.methodType(
-                    Object.class,
+                    Object[].class,
                     Arrays.stream(this.getPatternBindings())
                             .map(PatternBinding::getType)
                             .toArray(Class[]::new)
             );
-            MethodHandle initializingConstructor = SharedSecrets.getJavaLangRuntimeAccess().initializingConstructor(bindingMT);
+            MethodHandle pack = MethodHandles.identity(Object[].class).asCollector(Object[].class, this.getPatternBindings().length).asType(bindingMT);
```

Comment:
It's actually possible to avoid having to create a new MH each invocation, and instead have a hoisted, re-usable, handle.

```
private static Object[] collect(Object... params) {
    return params;
}

static final MethodHandle COLLECT_TO_ARRAY = MhUtil.findStatic(This.class, "collect", MethodType.methodType(Object[].class, Object[].class));
```